bitbucket.go

```diff
@@ -27,27 +27,37 @@ type Bitbucket struct {
 	} `json:"repository"`
 }
 
-func bitbucketToPayload(r io.Reader) *Payload {
+func bitbucketToPayload(r io.Reader, hookEvent string) (*Payload, error) {
 	var b Bitbucket
 	err := json.NewDecoder(r).Decode(&b)
 
 	if err != nil {
-		log.Printf("[ERR ] %s", err)
-		return nil
+		return nil, err
 	}
 
-	if len(b.Push.Changes) == 0 {
-		log.Println("[ERR ] Could not decode body into Payload")
-		return nil
+	// FIXME : remove this event name modification in future realease,
+	// made to avoid breaking change on 2022-05-20.
+	if hookEvent == "repo:push" {
+		hookEvent = "push"
+	}
+
+	var ref string
```

Comment:
Even tho I like the Var very much (83 represent), I think you could optimize this a bit by setting 'ref' directly to ""

Thus avoiding the else line 50

Reply:
Done, even tho I prefer the use of final variable, as in modern programming language like Java, where you can do thing such as:
```
final String ref;
if(tadam) { ...}
else if(youpla) { ...}
else { ref=""; }
```
And the compiler raises an error if the variable may not be initialized, or is initialized more than once.

Reply:
I can't believe other languages bother with boring names such as `const` 😑 when you could have epic ones like `final`🔥

On a serious note though, unfortunately in Go there is no immutable var declaration available. You don't have to always be immutable tho, but in case you really need immutable var, you could build something with a struct using generics with a private field 🙃

Reply:
I agree with you, can you imagine something like `const Fantasy` instead of `final Fantasy` ? No way.